src/overlays/actors/ovl_En_Stop_heishi/z_en_stop_heishi.c

```diff
@@ -32,45 +46,538 @@ ActorInit En_Stop_heishi_InitVars = {
+    } else {
+        EnStopHeishi_UpdateHeadThirdDay(this);
+    }
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->rotYTarget, 1, 0x1388, 0);
+
+    sp32 = 0;
+    if (gSaveContext.save.isNight) {
+        sp32 = 0x20;
+    }
 
-extern ColliderCylinderInit D_80AE88B0;
+    phi_a2 = 0;
+    switch (player->transformation) {
+        case PLAYER_FORM_HUMAN:
+            if ((STOLEN_ITEM_1 == SLOT_SWORD_GREAT_FAIRY) || (STOLEN_ITEM_1 == ITEM_SWORD_KOKIRI) ||
+                (STOLEN_ITEM_1 == ITEM_SWORD_RAZOR) || (STOLEN_ITEM_1 == ITEM_SWORD_GILDED) ||
+                (STOLEN_ITEM_2 == SLOT_SWORD_GREAT_FAIRY) || (STOLEN_ITEM_2 == ITEM_SWORD_KOKIRI) ||
+                (STOLEN_ITEM_2 == ITEM_SWORD_RAZOR) || (STOLEN_ITEM_2 == ITEM_SWORD_GILDED)) {
+                if (this->unk_265 != 0) {
+                    phi_a2 = 1;
+                }
+            } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_12_20)) {
+                phi_a2 = 1;
+            }
+            break;
 
-extern UNK_TYPE D_06006C18;
+        case PLAYER_FORM_DEKU:
+            phi_a2 = 4;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/EnStopheishi_Init.s")
+        case PLAYER_FORM_ZORA:
+            phi_a2 = 3;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/EnStopheishi_Destroy.s")
+        case PLAYER_FORM_GORON:
+            phi_a2 = 2;
+            break;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE750C.s")
+    if (((phi_a2 == 1) || (phi_a2 == 2) || (phi_a2 == 3)) &&
+        (((this->currentAnim == SOLDIER_ANIM_4)) || (this->currentAnim == SOLDIER_ANIM_5) ||
+         (this->currentAnim == SOLDIER_ANIM_6))) {
+        EnStopHeishi_ChangeAnim(this, SOLDIER_ANIM_STAND_HAND_ON_HIP);
+        if ((gSaveContext.save.day != 3) && gSaveContext.save.isNight) {
+            EnStopHeishi_ChangeAnim(this, SOLDIER_ANIM_STAND_LOOK_DOWN);
+        }
+        if (gSaveContext.save.day == 3) {
+            EnStopHeishi_ChangeAnim(this, SOLDIER_ANIM_STAND_HAND_ON_CHEST);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE75C8.s")
+    if ((phi_a2 == 0) || (phi_a2 == 4)) {
+        xDiff = this->actor.home.pos.x - player->actor.world.pos.x;
+        zDiff = this->actor.home.pos.z - player->actor.world.pos.z;
+        this->collider.dim.radius = 50;
+        this->collider.dim.height = 260;
+        if (sqrtf(SQ(xDiff) + SQ(zDiff)) < 240.0f) {
+            func_80AE795C(this, play);
+        } else {
+            func_80AE77D4(this);
+        }
+    } else {
+        this->collider.dim.radius = 15;
+        this->collider.dim.height = 60;
+    }
+    if (phi_a2 != 4) {
+        this->messageIndex = this->unk_276 * 8;
+        this->messageIndex += sp32;
+        this->messageIndex += phi_a2 * 2;
+    } else {
+        this->messageIndex = 0x40;
+        if (sp32 != 0) {
+            this->messageIndex = 0x48;
```

Comment:
Since this is only used as an index into an array, seems like it should be dec? Applies to both messageIndex and sp32, but if you think there's more structure to the arrays it'd be good to annotate them.